test/plugins/s3-persistence.spec.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2018 Rain Agency <contact@rain.agency>
+ * Author: Rain Agency <contact@rain.agency>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+"use strict";
+
+import { RequestEnvelope } from "ask-sdk-model";
+import { S3 } from "aws-sdk";
+import { expect } from "chai";
+import * as simple from "simple-mock";
+import { AlexaPlatform, plugins, VoxaApp, VoxaEvent } from "../../src";
+import { AlexaRequestBuilder } from "../tools";
+import { variables } from "../variables";
+import { views } from "../views";
+
+const rb = new AlexaRequestBuilder("user-xyz");
+
+describe("S3Persistence plugin", () => {
```

Comment:
Would be great if we did the testing using something like https://github.com/jamhall/s3rver